### ISSUE TITLE
Consider using Freenginx

### ISSUE THREAD

**Issue by tukusejssirs** [JonasAlfredsson/docker-nginx-certbot#275]
I would be interested in using Freenginx instead of Nginx.

@JonasAlfredsson, what do you think? :thinking:

Anyway, thanks for creating and maintaining `docker-nginx-certbot`! :pray:

**Comment by JonasAlfredsson** [comment]
Yeah, I have also thought about building the [unprivileged image of Nginx](https://hub.docker.com/r/nginxinc/nginx-unprivileged) as well.

But I have not had time to think about how this would be realized. 
Do we add more Dockerfiles into this repo? 
Do we move the /src/scrips folder to its own "nginx-certbot-utils" repo and make it a submodule to a "docker-freenginx-certbot" repo?

**Comment by tukusejssirs** [comment]
> Yeah, I have also thought about building the [unprivileged image of Nginx](https://hub.docker.com/r/nginxinc/nginx-unprivileged) as well.

Yeah, that would be awesome too, however, IMHO while that is a related feature, it is a different one. YMMV

> But I have not had time to think about how this would be realized.

At least for now, Freenginx should be compatible with Nginx, therefore, IMHO all that should be needed to do is (more or less) replacing Nginx with Freenginx. It might be different in the future, so you might want to think about this from maintenance POV too.

> Do we add more Dockerfiles into this repo?

Yes and you might want to publish it to Docker Hub, probably using a different (base) tag. That would also add up some extra work.

> Do we move the /src/scrips folder to its own "nginx-certbot-utils" repo and make it a submodule to a "docker-freenginx-certbot" repo?

I have no idea, maybe not. :wink:

**Comment by tukusejssirs** [comment]
This issue is also related ot #175, as Freenginx seem to have incorporated the experimental HTTP/3 support. Also look at [this Dockerfile](https://github.com/ammnt/freenginx/blob/main/Dockerfile). (Note that I am also interested in the QUIC / HTTP/3 support in `docker-nginx-certbot`.)

**Comment by JonasAlfredsson** [comment]
So one would have to build freenginx from source, and set up checks to keep it up to date, and then upload to DockerHub for us to then be able to include certbot+scripts. So it is definitely more work to create a "drop-in" replacement. 

Also, since this repo follows the main Nginx container, HTTP3 support is dictated by them.

**Comment by tukusejssirs** [comment]
> So one would have to build freenginx from source, and set up checks to keep it up to date, and then upload to DockerHub for us to then be able to include certbot+scripts. So it is definitely more work to create a "drop-in" replacement.

That makes sense.

> Also, since this repo follows the main Nginx container, HTTP3 support is dictated by them.

Actually, it works as expected (I have tested it with `docker-nginx-certbot`), however, at first I have forgotten to add `0.0.0.0:443:443/udp` to the exposed `ports` in my Docker Compose file. :wink: Therefore, that is not an issue anymore.

We could document it though. I could create a PR with what I did to make it work. Where would you like to at this? To [`docs/advanced_usage.md`](https://github.com/JonasAlfredsson/docker-nginx-certbot/blob/master/docs/advanced_usage.md)? Or should I rather create a new file (e.g. `docs/enable_http3_quic.md`)?

**Comment by JonasAlfredsson** [comment]
What do you want to document? 
The HTTP3 stuff is not really part of the nginx-certbot codebase, so it might be misplaced in this repo. But would you mind pasting what you intended to write here and we can see if it does fit? :)
